src/patients/allergies/NewAllergyModal.tsx

```diff
@@ -14,6 +14,8 @@ const NewAllergyModal = (props: NewAllergyModalProps) => {
   const { show, onCloseButtonClick, onSave } = props
   const [allergy, setAllergy] = useState({ name: '' })
   const [errorMessage, setErrorMessage] = useState('')
+  const [isInvalid, setIsInvalid] = useState(false)
```

Comment:
Let's rename this field to `isAllergyNameInvalid` to better reflect the usage of this variable.

The setter should then become `setIsAllergynameInvalid`